bolt/include/bolt/Core/BinaryFunction.h

```diff
@@ -428,6 +428,9 @@ class BinaryFunction {
   /// Function order for streaming into the destination binary.
   uint32_t Index{-1U};
 
+  /// Function is referenced by a non-control flow instruction.
```

Comment:
Add a comment that at the moment this flag is only valid under ICF. When we'll need it for other purposes, we should refactor the code to make it generally available.